firestore/integration_test_internal/src/firestore_integration_test.h

```diff
@@ -355,6 +350,24 @@ class FirestoreIntegrationTest : public testing::Test {
   mutable std::unordered_map<Firestore*, FirestoreInfo> firestores_;
 };
 
+template <typename PredT>
+bool WaitUntil(const PredT& pred, int timeout_ms) {
+  auto now = std::chrono::steady_clock::now();
```

Comment:
The logic in this method is a bit difficult to follow IMO because `now` changes value and meaning over time. Consider getting rid of the `now` variable and just calling `std::chrono::steady_clock::now()` when the current time is needed.

I realize that `now` is used to determine the final result on line 368, but this could be avoided if the `while` loop were modified to be `while (true)`, something like this:

```
while (true) {
  if (pred()) {
    return true;
  } else if (std::chrono::steady_clock::now() >= timeout_time) {
    return false;
  }
  ...
}
```

I think a change like this would improve the overall readability of the method.

Reply:
I usually try to avoid `while (true)` loops where possible -- personally, it feels like the loop simply obscures the loop condition (effectively there still is a loop condition, it's just implicit in the loop body).

If you feel strongly about getting rid of `now`, the last line could instead be something like `return pred();`. What do you think?

Alternatively, perhaps renaming `now` could help? Maybe something like `last_checked_time`?